src/main/kotlin/graphics/scenery/volumes/VDIVolumeManager.kt

```diff
@@ -0,0 +1,200 @@
+
+        val lowestLevel = MemoryUtil.memCalloc(numGridCells.x.toInt() * numGridCells.y.toInt() * numGridCells.z.toInt() * 4)
+
+        val outputSubVDIColor: Texture = Texture.fromImage(
+            Image(outputSubColorBuffer, maxSupersegments, windowHeight, windowWidth, FloatType()), usage = hashSetOf( Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture),
+            type = FloatType(), channels = 4, mipmap = false,  normalized = false, minFilter = Texture.FilteringMode.NearestNeighbour, maxFilter = Texture.FilteringMode.NearestNeighbour)
+        volumeManager.customTextures.add("OutputSubVDIColor")
+        volumeManager.material().textures["OutputSubVDIColor"] = outputSubVDIColor
+
+        val outputSubVDIDepth: Texture = Texture.fromImage(
+            Image(outputSubDepthBuffer, 2 * maxSupersegments, windowHeight, windowWidth, FloatType()),  usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture),
+            type = FloatType(), channels = 1, mipmap = false, normalized = false, minFilter = Texture.FilteringMode.NearestNeighbour, maxFilter = Texture.FilteringMode.NearestNeighbour)
+        volumeManager.customTextures.add("OutputSubVDIDepth")
+        volumeManager.material().textures["OutputSubVDIDepth"] = outputSubVDIDepth
+
+        val gridCells: Texture = Texture.fromImage(
+            Image(lowestLevel, numGridCells.x.toInt(), numGridCells.y.toInt(), numGridCells.z.toInt(), UnsignedIntType()), channels = 1, type = UnsignedIntType(),
+            usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture))
+        volumeManager.customTextures.add("OctreeCells")
+        volumeManager.material().textures["OctreeCells"] = gridCells
+
+        volumeManager.customUniforms.add("doGeneration")
+        volumeManager.shaderProperties["doGeneration"] = true
+
+        val compute = RichNode()
+        compute.setMaterial(ShaderMaterial(Shaders.ShadersFromFiles(arrayOf("GridCellsToZero.comp"), this::class.java)))
+
+        compute.metadata["ComputeMetadata"] = ComputeMetadata(
+            workSizes = Vector3i(numGridCells.x.toInt(), numGridCells.y.toInt(), 1),
+            invocationType = InvocationType.Permanent
+        )
+
+        compute.material().textures["GridCells"] = gridCells
+
+        scene.addChild(compute)
+
+        return volumeManager
+    }
+
+    private fun vdiCompact(windowWidth: Int, windowHeight: Int, maxSupersegments: Int, scene: Scene, hub: Hub): VolumeManager
+    {
+        val raycastShader = "AdaptiveVDIGenerator.comp"
+        val accumulateShader = "AccumulateVDI.comp"
+
+        val volumeManager = instantiateVolumeManager(raycastShader, accumulateShader, hub)
+
+        val totalMaxSupersegments = maxSupersegments * windowWidth * windowHeight
+
+        val outputSubColorBuffer = MemoryUtil.memCalloc(512 * 512 * ceil((totalMaxSupersegments / (512*512)).toDouble()).toInt() * 4 * 4)
```

Comment:
1969fec72a164260db177ee1cd96a4007893a92c adds a `close()` function, but I am not sure how this should be invoked. The `VolumeManager` created by the `VDIVolumeManager` can be expected to have the lifetime of the application and therefore the buffers need to have the same lifetime too. Is there a routine that runs on application shutdown where this could be called?